### ISSUE TITLE
[BUG] - Cannot read properties of undefined (reading 'fields')

### ISSUE THREAD

**Issue by zbeyens** [premieroctet/next-admin#268]
### Description

I'm getting this issue when visiting any model "Edit" page.

### Reproduction steps

```ts
async function NextAdminPage({ params, searchParams }: PageProps) {
  const props = await getPropsFromParams({
    action: submitFormAction,
    deleteAction: deleteItemAction,
    options: adminOptions,
    params: params.nextadmin,
    prisma,
    schema,
    searchPaginatedResourceAction,
    searchParams,
  });

  return <NextAdmin {...props} />;
}

// actions.ts
'use server';

import type { ActionParams, ModelName } from '@premieroctet/next-admin';

import {
  type SearchPaginatedResourceParams,
  deleteResourceItems,
  searchPaginatedResource,
  submitForm,
} from '@premieroctet/next-admin/dist/actions';

import { adminOptions } from '@/app/(dynamic)/(main)/studio/admin/[[...nextadmin]]/admin-options';
import { prisma } from '@/lib/db/prisma';

export const submitFormAction = async (
  params: ActionParams,
  formData: FormData
) => {
  return submitForm({ ...params, options: adminOptions, prisma }, formData);
};

export const deleteItemAction = async (
  model: ModelName,
  ids: number[] | string[]
) => {
  return deleteResourceItems(prisma, model as any, ids);
};

export const searchPaginatedResourceAction = async (
  actionParams: ActionParams,
  params: SearchPaginatedResourceParams
) => {
  return searchPaginatedResource(
    { ...actionParams, options: adminOptions, prisma },
    params
  );
};

// next config
// ...
experimental: {
    swcPlugins: [
      [
        "next-superjson-plugin",
        {
          excluded: [],
        },
      ],
    ],
    // typedRoutes: true
  },
// ...

// @prisma/client 5.13.0
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextSearch", "fullTextIndex", "driverAdapters"]
}

generator jsonSchema {
  provider              = "prisma-json-schema-generator"
  includeRequiredFields = "true"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

generator kysely {
  provider        = "prisma-kysely"
  previewFeatures = ["fullTextSearch", "fullTextIndex"]
  output          = "./kysely"
}

generator types {
  provider = "prisma-type-generator"
  output   = "../src/lib/db"
}

model ShowGenre {
  id      String @id @default(cuid())
  showId  String
  Show    Show   @relation(fields: [showId], references: [id], onDelete: Cascade)
  genreId String
  Genre   Genre  @relation(fields: [genreId], references: [id], onDelete: Cascade)

  @@unique([showId, genreId])
}
// ...
```

### Next router

App router

### Next Admin version

4.1.0

### Next Admin options

```ts
{
  basePath: '/studio/admin',
  model: {},
}
```

### Logs

```bash
Uncaught TypeError: Cannot read properties of undefined (reading 'fields')
    at eval (webpack-internal:///(rsc)/./node_modules/.pnpm/@premieroctet+next-admin@4.1.0_@babel+core@7.24.5_@prisma+client@5.13.0_@types+react-dom@18.3_742vsuvunpgktjnkmrcwvfoehq/node_modules/@premieroctet/next-admin/dist/utils/props.js:117:43)
    at Array.forEach (<anonymous>)
    at getPropsFromParams (webpack-internal:///(rsc)/./node_modules/.pnpm/@premieroctet+next-admin@4.1.0_@babel+core@7.24.5_@prisma+client@5.13.0_@types+react-dom@18.3_742vsuvunpgktjnkmrcwvfoehq/node_modules/@premieroctet/next-admin/dist/utils/props.js:110:40)
    at async NextAdminPage (webpack-internal:///(rsc)/./src/app/(dynamic)/(main)/studio/admin/[[...nextadmin]]/page.tsx:52:19)
```

### Browsers

Chrome

**Comment by nahtnam** [comment]
I'm having a similar issue where I can't load the admin panel at all.

Here is the client error:

```
Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'User')

Call Stack
useDataColumns
node_modules/@premieroctet/next-admin/dist/hooks/useDataColumns.js (11:1)
List
node_modules/@premieroctet/next-admin/dist/components/List.js (33:36)
```

<details><summary>and server error</summary>
<p>

```
Internal error: TypeError: Cannot read properties of undefined (reading 'User')
    at useDataColumns (./node_modules/@premieroctet/next-admin/dist/hooks/useDataColumns.js:23:40)
    at List (./node_modules/@premieroctet/next-admin/dist/components/List.js:58:86)
    at au (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:10446)
    at /my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:15122
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:16908)
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:46505)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at aj (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:53793)
    at /my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:42336
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:42822)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at aj (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:53793)
    at aC (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:52874)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:50095)
    at aj (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:53793)
    at /my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:42336
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:42822)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at aj (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:53793)
    at aC (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:52874)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:50095)
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:45321)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at ab (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:11808)
    at /my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:16871
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:16908)
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:46505)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:45321)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at ab (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:11808)
    at /my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:16871
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:16908)
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:46505)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:45321)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at ab (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:11808)
    at /my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:16871
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:16908)
    at aw (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:46505)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:49776)
    at aj (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:53793)
    at aC (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:52874)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:50095)
    at a_ (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:50067)
    at /my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:58612
    at /my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:59270
    at aN (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:59278)
    at Timeout._onTimeout (/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:6939)
digest: "2087400674"
```

</p>
</details> 

Downgrading to 4.0.0 fixes it, 4.1.0 is broken

**Comment by zbeyens** [comment]
I confirm all the pages work on 4.0.0.

There is just one runtime error left: 

```bash
Uncaught TypeError: Cannot read properties of null (reading 'parentNode')
    at $RS (vote:1:32755)
    at vote:312:59529
```

For such schema:

```prisma
model Vote {
  id      String  @id @default(cuid())
  userId  String
  User    User    @relation(fields: [userId], references: [id], onDelete: Cascade)
  showId  String?
  Show    Show?   @relation(fields: [showId], references: [id], onDelete: Cascade)
  mediaId String?
  Media   Media?  @relation(fields: [mediaId], references: [id], onDelete: Cascade)

  value Int @default(1)

  @@unique([userId, showId])
  @@unique([userId, mediaId])
}
```

**Comment by cregourd** [comment]
This bug is fixed in [v4.2.0](https://www.npmjs.com/package/@premieroctet/next-admin/v/4.2.0) 🚀

Feel free to give us feedback
